assets/chat/js/menus.js

```diff
@@ -241,21 +225,38 @@ class ChatUserMenu extends ChatMenu {
             const searching = this.searchterm.length > 0;
             if(searching && this.totalcount !== this.searchcount) {
                 this.header.text(`Users (${this.searchcount} out of ${this.totalcount})`);
+                [...this.sections.values()].forEach((section) => {
+                    $(section.title).text(`${section.searchcount} out of ${section.users.children.length} ${section.label}${section.users.children.length === 1 ? '' : 's'}`);
+                    if (section.searchcount === 0) $(section.container).hide();
+                    else $(section.container).show();
+                });
             } else {
                 this.header.text(`Users (${this.totalcount})`);
+                [...this.sections.values()].forEach((section) => {
+                    $(section.title).text(`${section.users.children.length} ${section.label}${section.users.children.length === 1 ? '' : 's'}`);
+                    if (section.users.children.length === 0) $(section.container).hide();
+                    else $(section.container).show();
+                });
             }
             this.ui.toggleClass('search-in', searching);
         }
         super.redraw();
     }
 
     addAll(){
-        this.totalcount = 0;
-        this.container.empty();
-        [...this.chat.users.keys()].forEach(username => this.addElement(username));
-        this.sort();
-        this.filter();
-        this.redraw();
+        if (this.hasFlairs) {
+            this.totalcount = 0;
+            this.container.empty();
+            this.sections = new Map();
+            [...this.chat.flairsMap.values(), {label: 'User', name: 'user', priority: 9, color: '#4A8ECC'}]
+            .sort((a, b) => a.priority - b.priority)
+            .filter((a) => a.priority <= 10 && a.name !== 'flair20') // 10 or lower and not Verified.
```

Comment:
I acknowledge that creating a section for every single flair is overwhelming, especially if some flairs only belong to a few users, but is there any particular reason you used this logic to select the flairs? This excludes some flairs that I think deserve distinction (such as Contributor and Moderator), and includes a one-off flair that absolutely does not (NFL Andy). Additionally, it's a little fragile in that changing priorities can unexpectedly add/create new sections, and is tailored to Destiny.gg specifically.

I personally think the best approach is to only create sections using a subset of the flairs defined [in our seed script](https://github.com/destinygg/website/blob/master/config/destiny.gg.data.sql#L19-L41). These are guaranteed (probably 😅) to always exist in Destiny.gg and any clone websites. I also recommend ordering sections manually rather than relying on each flair's priority.

What do you think?

Reply:
Yeah, I wanted to make it as dynamic as I could so adding a new user list section would be just making a new flair.
I didn't know how public the moderators wanted to be since they didn't have a name colour or a badge. So these were the parameters I found worked best for the list.

I do think a static list would work better.

Let me know your thoughts of this list (in order).
Unsure if Trusted should be on the list.
Unsure of Contributors position in the order and if it should include emote contributors which is not currently in the seed script.

Admins
Moderators
Broadcasters
Vips
Trusted
Contributors
Subscriber Tier 4
Subscriber Tier 3
Subscriber Tier 2
Subscriber Tier 1
Users
Bots

Reply:
I think your list is great. Let's go with that.

Reply:
@11k What are your thoughts on flair icons?
![image](https://user-images.githubusercontent.com/13901269/199680242-7897fdd6-52a7-4f3f-b8ea-4ea204a17578.png)

Reply:
I think it looks a little too busy when people have multiple flairs. I also like it better when all the names are left-aligned. How about just the icon for each flair (if there is one) to the left of the section title?

Reply:
@11k 
![image](https://user-images.githubusercontent.com/13901269/199859285-94a7b72e-51ca-473e-a35a-a6d69b901865.png)

Reply:
Combining both emote and regular contributors in the same section is a nice touch.

My only request is you keep the section titles and dividers gray like in your last screenshot. The headers kind of blend together with usernames if they're colored and my brain has to work harder to process what I see.